packages/swingset-liveslots/src/vpid-tracking.md

```diff
@@ -2,85 +2,88 @@
 
 Kernels and vats communicate about promises by referring to their VPIDs: vat(-centric) promise IDs. These are strings like `p+12` and `p-23`. Like VOIDs (object IDs), the plus/minus sign indicates which side of the boundary allocated the number (`p+12` and `o+12` are allocated by the vat, `p-13` and `o-13` are allocated by the kernel). But where the object ID sign also indicates which side "owns" the object (i.e. where the behavior lives), the promise ID sign is generally irrelevant.
 
-Instead, we care about which side holds the resolution authority. This is not indicated by the VPID sign, and in fact is not necessarily static. Liveslots does not currently have any mechanism to allow one promise to be forwarded to another, but if it acquires this some day, then the decider of a promise could shift from kernel to vat to kernel again before it finally gets resolved. And there *is* a sequence that allows a vat to receive a reference to a promise in method arguments first, then receive a message whose result promise uses that VPID second (`p1 = p2~.foo(); x~.bar(p1)`, then someone resolves `p2` to `x`). In this case, the decider is initially the kernel, then the authority is transferred to the vat later.
+Instead, we care about which side holds the resolution authority for a promise (referred to as being its **decider**). This is not indicated by the VPID sign, and in fact is not necessarily static. Liveslots does not currently have any mechanism to allow one promise to be forwarded to another, but if it acquires this some day, then the decider of a promise could shift from kernel to vat to kernel again before it finally gets resolved. And there *are* sequences that allow a vat to receive a reference to a promise in method arguments before receiving a message for which that same VPID identifies the result promise (e.g., `pk = makePromiseKit(); p2 = E(pk.promise).push('queued'); await E(observer).push(p2); pk.resolve(observer)` with an observer whose `push` returns a prompt response). In such cases, the decider is initially the kernel but later becomes the receiving vat.
```

Comment:
Here's my understanding of how decidership of p2 changes; please let me know if I've got anything wrong.
```
step1: pk = makePromiseKit();
// pk.promise is local to Alice

step2: p2 = E(pk.promise).push('queued');
// syscall.send(target=pk.promise, method='push', args=['queued'], result=p2})
// Kernel becomes the decider of p2 and notes that Alice is the decider of pk.promise

step3: await E(observer).push(p2);
// syscall.send(target=observer, method='push', args=[p2], result=<anonymous>)
// Kernel becomes the decider of <anonymous> and notes subscription by Alice
// Alice loses agency

// Kernel handles the step2 syscall.send by pushing onto the queue for pk.promise

// Kernel handles the step3 syscall.send by
//   deliver(type='send', target=observer, method='push', args=[p2], result=<anonymous>)
//   and making observer's vat the decider of <anonymous>
// observer does syscall.resolve(subject=<anonymous>, resolution=…)
// observer loses agency

// Kernel handles the syscall.resolve, which involves
//   notify(vatID=Alice, promise=<anonymous>) to enqueue a message

// Kernel handles the notify by
//   deliver(type='notify', subject=<anonymous>, resolution=…)
// Alice regains agency
step4: pk.resolve(observer)
// syscall.resolve(subject=pk.promise, resolution=observer)
// Alice loses agency

// Kernel handles the step4 syscall.resolve, which involves
//   noting resolution of pk.promise to observer and
//   requeueKernelPromise(pk.promise) to requeue messages to it

// Kernel handles the requeued step2 syscall.send by
//   deliver(type='send', target=observer, method='push', args=['queued'], result=p2)
//   and making observer's vat the decider of p2
//   and (hopefully!) also noting subscription by Alice to p2
```

Reply:
> step2: p2 = E(pk.promise).push('queued');
>    syscall.send(target=pk.promise, method='push', args=['queued'], result=p2)

Ah, ok since `pk.promise` is local, this E() is handled entirely by HandledPromise, and liveslots doesn't see it at all, until/unless `pk.promise` gets resolved to something that liveslots created.

(I'm betting that invalidates the rest of the sequence, so I didn't study it carefully yet, we can sync up on this in realtime and work out an example that demonstrates the problem, or maybe remove it from the docs.. I'm pretty sure I wrote the original to remind myself in the future, because it wasn't easy to reconstruct the problematic scenario, so it'd be nice to include at least a few hints)